Backend/Backend.WebServices/Controllers/EnterpriseController.cs

```diff
@@ -51,11 +51,26 @@ public IEnumerable<EnterpriseDTO> Get()
         /// Given an Id (and possibly a password), get details of a specific enterprise
         /// </summary>
         /// <param name="id">ID of the enterprise whose details will be returned</param>
-        /// <returns>A string representing an enterprise</returns>
-        public string Get(int id)
+        /// <param name="password">Password of the enterprise</param>
+        /// <returns>An EnterpriseDTO with the specified ID</returns>
+        public EnterpriseDTO Get(int id, string password = null)
         {
             ThrowExceptionIfNotHttps();
-            return "ForestedgeCommunityGarden";
+            var enterprises = _context.Enterprises.ToList();
+            Enterprise enterprise = null;
+
+            foreach (Enterprise e in enterprises)
+                if (e.Id == id)
+                    enterprise = e;
+
+            if (password != null && enterprise.Password != null)
+                if (!enterprise.Password.Equals(password))
+                    throw new HttpException(403, "Password is incorrect");
+
+            if (enterprise == null)
+                throw new HttpException(404, "Enterprise not found");
+
+            return new EnterpriseDTO(enterprise);
```

Comment:
The EnterpriseDTO is not a good DTO to use here; it includes a very slimmed-down version of the enterprise that is only suitable for showing a list of enterprises. Instead we want to return a new DTO that includes all the details of the enterprise, including participants, places, etc.

I'd suggest making a class called EnterpriseDetailDTO, along with the classes ParticipantDTO, PlaceDTO, etc. You'd then need to write some code in each of these to map the Database classes to the DTOs.